architecture/2024-08-dashboard-component-threat-model.md

```diff
@@ -0,0 +1,299 @@
+
+Users that are signed-in in to Backstage generally have full access to all information and actions. If more fine-grained control is required, the permissions system should be enabled and configured to restrict access as necessary.
+
+This web application is not intended to be public-facing; it is available on the intranet for use by both development and operations personnel working on a radified application using kubernetes port-forward
+```
+kubectl port-forward --namespace=radius-system svc/dashboard 7007:80
+``` 
+
+Decisions to make Dashboard public-facing should be the user's conscious choice. 
+
+
+
+## Assumptions
+
+This threat model assumes that:
+
+1. The Radius installation is not tampered with.
+2. The Kubernetes cluster that Radius is installed on is not compromised.
+3. It is the responsibility of the Kubernetes cluster to authenticate users. Administrators and users with sufficient privileges can perform their required tasks. Radius cannot prevent actions taken by an administrator.
+4. Dashboard users have been configured to have right level of access by following [Backstage Threat Model](https://backstage.io/docs/overview/threat-model/). These users are trusted to the extent that they are not expected to compromise the availability of Backstage, but they are not trusted to not compromise data confidentiality or integrity.
+5. Dashboard is not public facing.
+6. Access to Dashboard is using HTTPS.
+7. Authentication mechanism provided by Backstage is robust.
+
+## Data Flow
+
+### Diagram
+
+![Radius Dashboard](2024-08-dashboard-component-threat-model/dashboard-tm.png)
+
+1. User types the backstage url and accesses Radius plugin
+2. Request reaches the dashboard pod in `radius-system` namespace in kubernetes cluster.
+3. The dashboard service sends a Radius API request to UCP.
+4. UCP works with ApplicationsCore-RP and sends response back to Dashboard SPA.
+5. Dashboard SPA contructs the visuals using components from backstage core, rad-component and data in API response and responds with appropriate page to the user. 
+
+### Threats
+ 
+#### Threat 1: DoS
+
+**Description**
+
+A client can access Dashboard repeatedly or fetch the page in a loop.
+
+**Impact**
+
+Due to the volume of requests Dashboard as well as the UCP, AppCore-RP components involved in serving the request could run out of resource to serve a legitimate request.
+   
+**Mitigation**:
+
+1. Access to Dashboard should be provided to trusted users.  The [Backstage  permissions system](https://backstage.io/docs/permissions/overview) should be enabled and configured to restrict access as necessary.
+   
+**Status**:
+
+Active. Operators are expected to configure the system and limit access to Dashboard portal. 
+
+2. Audit logs should be enabled to monitor and report on suspicious user activity. 
+
+**Status**
+
+[In Progress](https://github.com/backstage/backstage/issues/23950)
```

Comment:
Will this be enabled by default or opt in?

Reply:
Follow up once the feature is implemented and make a decision.

Reply:
created https://github.com/radius-project/dashboard/issues/105 to address the documentation which captures the remediation/ future revisit.